### DISCUSSION TITLE
GraphQL API Retrieve event history text of issue

### DISCUSSION THREAD

**Discussion by Blackglade** [community/community#24437] [Apps, API and Webhooks]
<p>I have the following query:</p>
<pre><code>query issues($org: String!, $repo: String!) {
  repository(owner: $org, name: $repo) {
    issues(first: 77){
      nodes{
        number
        timelineItems(itemTypes:[ADDED_TO_PROJECT_EVENT, MOVED_COLUMNS_IN_PROJECT_EVENT], first: 100){
          nodes {
            __typename
            ... on AddedToProjectEvent{
              id
              actor {
                login
              }
            }
          }
        }
      }
    }
  }
}
</code></pre>
<p>I basically have a project board setup with several columns and want to capture the issues as they move throughout the column. How do I retrieve the event history of what columns each moves to and from. This query will only give me the event and the id/author of the event, but not details about the event (ie: which column in the project board it was moved to).</p>

**Comment by lee-dohm** [comment, marked as answer]
<p>You can use the following query:</p>
<pre><code>query issues($org: String!, $repo: String!) {
  repository(owner: $org, name: $repo) {
    issues(last: 100){
      nodes{
        number
        timelineItems(itemTypes:[ADDED_TO_PROJECT_EVENT, MOVED_COLUMNS_IN_PROJECT_EVENT], first: 100){
          nodes {
            __typename
            ... on AddedToProjectEvent{
              id
              actor {
                login
              }
            }
            ... on MovedColumnsInProjectEvent {
              id
              previousProjectColumnName
              projectColumnName
            }
          }
        }
      }
    }
  }
}
</code></pre>
<p>As stated in the documentation, you have to <a href="https://developer.github.com/v4/previews/#project-event-details" rel="nofollow noopener">enable the preview feature</a> for those particular fields to be available on <code>MovedColumnsInProjectEvent</code>.</p>
<p>I hope that helps!</p>

**Reply by mseankelly-Kin** [reply]
This query appears to not work for new projects (non-classic). See: [#30979](https://github.com/community/community/discussions/30979)

**Comment by Blackglade** [comment]
<p>Thanks this helps a lot!</p>
